web/src/common/CaptchaPreview.js

```diff
@@ -100,24 +110,40 @@ export const CaptchaPreview = ({ provider, providerName, clientSecret, captchaTy
     setCaptchaToken(token);
   };
 
-
   const renderCheck = () => {
     if (captchaType === "Default") {
       return renderDefaultCaptcha();
     } else {
       return (
-        <CaptchaWidget
-          captchaType={captchaType}
-          siteKey={clientId}
-          onChange={onSubmit}
-        />
+        <Col>
+          <Row>
+            <CaptchaWidget
+              captchaType={captchaType}
+              subType={subType}
+              siteKey={clientId}
+              clientSecret={secret}
+              onChange={onSubmit}
+              clientId2={clientId2}
+              clientSecret2={secret2}
+            />
+          </Row>
+        </Col>
       );
     }
   };
 
   return (
     <React.Fragment>
-      <Button style={{ fontSize: 14 }} type={"primary"} onClick={clickPreview} disabled={captchaType !== "Default" && (!clientId || !clientSecret)}>
+      <Button
+        style={{ fontSize: 14 }}
+        type={"primary"}
+        onClick={clickPreview}
+        disabled={
+          (captchaType !== "Default" && (!clientId || !clientSecret)) ||
```

Comment:
this logic is too complicated, define `disabled` with multi-line if-else outside of the return.